web/src/composebox_typeahead.ts

```diff
@@ -954,6 +954,12 @@ export function get_candidates(
                     topic_list.push(token);
                 }
                 const matcher = get_topic_matcher(token);
+                // We want the empty topic's display name to match the query
+                // while searching for topics in the typeahead.
+                const empty_topic_idx = topic_list.indexOf("");
+                if (empty_topic_idx !== -1) {
+                    topic_list[empty_topic_idx] = util.get_final_topic_display_name("");
+                }
```

Comment:
This logic seems very strange. Why does it only adjust the name in non-first positions? I don't see any product reason for such behavior.

Reply:
I think this is fine. Tim maybe read `-1` as `1` or something. I'll wait for him to clarify.

I have few other suggestions:
* rename `topic` to `topic_display_name` in `TopicSuggestion` type. -- We are following a policy of calling the display_name as `topic_display_name` instead of `topic` as "test general chat" is not the actual topic value.
* Include `is_empty_string` in `TopicSuggestion`, there's no real benefit of calculating this later.
* If we think `topic` and `topic_display_name` both is required in the `.hbs` template, include both `topic` and `topic_display_name` in the `TopicSuggestion` type.